src/LockManagerERC20.sol

```diff
@@ -0,0 +1,46 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+pragma solidity ^0.8.13;
+
+import {LockManagerBase} from "./base/LockManagerBase.sol";
+import {ILockManager} from "./interfaces/ILockManager.sol";
+import {LockManagerSettings} from "./interfaces/ILockManager.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+/// @title LockManagerERC20
+/// @author Aragon X 2025
+/// @notice Helper contract acting as the vault for locked tokens used to vote on multiple plugins and proposals.
+contract LockManagerERC20 is ILockManager, LockManagerBase {
+    /// @notice The address of the token contract used to determine the voting power
+    IERC20 private immutable erc20Token;
+
+    /// @param _settings The operation mode of the contract (plugin mode)
+    /// @param _token The address of the token contract that users can lock
+    constructor(LockManagerSettings memory _settings, IERC20 _token) LockManagerBase(_settings) {
+        erc20Token = _token;
+    }
+
+    /// @inheritdoc ILockManager
+    /// @dev Not having `token` as a public variable because the return types would differ (address vs IERC20)
+    function token() public view virtual returns (address _token) {
+        return address(erc20Token);
+    }
+
+    // Overrides
+
+    /// @inheritdoc LockManagerBase
+    function _transfer(address _recipient, uint256 _amount) internal virtual override {
+        erc20Token.transfer(_recipient, _amount);
+    }
+
+    /// @inheritdoc LockManagerBase
+    function _lock() internal virtual override {
+        uint256 _allowance = erc20Token.allowance(msg.sender, address(this));
+        if (_allowance == 0) {
+            revert NoBalance();
```

Comment:
Do you need this check? Doesn't ERC20 allowance already check this in transferFrom?

Reply:
The point is not about detecting that you spend more that you have. The point here is determining how much money can be locked. If 0 tokens can be taken, we will be spending gas for no actual outcome.

Reply:
Oh I see, if allowance == 0, you will attempt a zero transfer